dashboard/app/models/level.rb

```diff
@@ -239,6 +239,14 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  # Find the first script this level belongs to that is a HOC script.
+  # Returns nil if this level does not belong to any HOC scripts.
+  def find_hoc_script
+    hoc_script_level = self.script_levels.find{|l| l.script.hoc?}
```

Comment:
it should be if the Level was loaded from the Script cache (which is true for ScriptLevelsController#show). For LevelSourcesController#show (ie. the /c/ page) it is not. We should fix the /c/ page in general to use a Levels cache...